src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -550,32 +551,31 @@ public ClusterEvent.IClusterDomainEvent Event
         }
     }
 
-    //TODO: IRequiresMessageQueue?
     /// <summary>
     /// Supervisor managing the different Cluster daemons.
     /// </summary>
-    internal sealed class ClusterDaemon : UntypedActor
+    internal sealed class ClusterDaemon : UntypedActor, IRequiresMessageQueue<IUnboundedMessageQueueSemantics>
     {
-        readonly IActorRef _coreSupervisor;
+        IActorRef _coreSupervisor;
         readonly ClusterSettings _settings;
 
         public ClusterDaemon(ClusterSettings settings)
         {
-            // Important - don't use Cluster(context.system) here because that would
+            // Important - don't use Cluster(context.system) in constructor because that would
             // cause deadlock. The Cluster extension is currently being created and is waiting
             // for response from GetClusterCoreRef in its constructor.
-            _coreSupervisor =
-                Context.ActorOf(Props.Create<ClusterCoreSupervisor>().WithDispatcher(settings.UseDispatcher), "core");
-
-            Context.ActorOf(Props.Create<ClusterHeartbeatReceiver>().WithDispatcher(settings.UseDispatcher), "heartbeatReceiver");
-
             _settings = settings;
         }
 
         protected override void OnReceive(object message)
         {
             message.Match()
-                .With<InternalClusterAction.GetClusterCoreRef>(msg => _coreSupervisor.Forward(msg))
+                .With<InternalClusterAction.GetClusterCoreRef>(msg =>
+                {
+                    if(_coreSupervisor == null)
```

Comment:
Moved the lazy initialization to here instead - avoid calling the `Cluster` object in the constructors of any actors who are needed for the cluster to startup.